google/cloud/aiplatform/metadata/metadata.py

```diff
@@ -15,161 +15,382 @@
+            "input:" from PipelineJob parameter keys
+
+        Returns:
+          Dict of custom properties with keys mapped to column names
+        """
+        column_key_to_value = {}
+        for key, value in metadata.items():
+            if filter_prefix and key.startswith(filter_prefix):
+                key = key[len(filter_prefix) :]
+            column_key_to_value[".".join([metadata_type, key])] = value
+
+        return column_key_to_value
+
+
+class _ExperimentTracker:
+    """Tracks Experiments and Experiment Runs wil high level APIs"""
+
     def __init__(self):
-        self._experiment = None
-        self._run = None
-        self._metrics = None
+        self._experiment: Optional[experiment_resources.Experiment] = None
+        self._experiment_run: Optional[experiment_run_resource.ExperimentRun] = None
 
     def reset(self):
-        """Reset all _MetadataService fields to None"""
+        """Resets this experiment tracker, clearing the current experiment and run."""
         self._experiment = None
-        self._run = None
-        self._metrics = None
+        self._experiment_run = None
 
     @property
     def experiment_name(self) -> Optional[str]:
-        """Return the experiment name of the _MetadataService, if experiment is not set, return None"""
+        """Return the currently set experiment name, if experiment is not set, return None"""
         if self._experiment:
-            return self._experiment.display_name
+            return self._experiment.name
         return None
 
     @property
-    def run_name(self) -> Optional[str]:
-        """Return the run name of the _MetadataService, if run is not set, return None"""
-        if self._run:
-            return self._run.display_name
-        return None
+    def experiment(self) -> Optional[experiment_resources.Experiment]:
+        "Returns the currently set Experiment."
+        return self._experiment
 
-    def set_experiment(self, experiment: str, description: Optional[str] = None):
-        """Setup a experiment to current session.
+    @property
+    def experiment_run(self) -> Optional[experiment_run_resource.ExperimentRun]:
+        """Returns the currently set experiment run."""
+        return self._experiment_run
+
+    def set_experiment(
+        self,
+        experiment: str,
+        *,
+        description: Optional[str] = None,
+        backing_tensorboard: Optional[
+            Union[str, tensorboard_resource.Tensorboard]
+        ] = None,
+    ):
+        """Set the experiment. Will retrieve the Experiment if it exists or create one with the provided name.
 
         Args:
             experiment (str):
-                Required. Name of the experiment to assign current session with.
+                Required. Name of the experiment to set.
             description (str):
                 Optional. Description of an experiment.
+            backing_tensorboard Union[str, aiplatform.Tensorboard]:
```

Comment:
Looks like we have option to add tensorboard for both the experiment setup and run setup. I wonder what is the desired behavior here if they are not the same?

Reply:
`Tensorboard` set at experiment run creation time will override the `Tensorboard` set at experiment creation time. I update `start_run` docstring with a comment describing this behavior.